go/vt/mysqlctl/mysqld.go

```diff
@@ -220,8 +220,14 @@ func (mysqld *Mysqld) startNoWait(ctx context.Context, cnf *Mycnf, mysqldArgs ..
 				return err
 			}
 		}
+		mysqlBaseDir, err := vtenv.VtMysqlBaseDir()
+		if err != nil {
+			return err
+		}
 		arg := []string{
-			"--defaults-file=" + cnf.path}
+			"--defaults-file=" + cnf.path,
+			"--basedir" + mysqlBaseDir,
```

Comment:
@morgo @sougou pretty sure you want an = after basedir here

Reply:
Thanks! It will be fixed in https://github.com/vitessio/vitess/pull/5027